src/services/transactions/trackServerTransactions.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  NotificationTypesEnum,
+  SendTransactionReturnType,
+  SignedTransactionType,
+  TrackTransactionsPropsType,
+  TransactionBatchStatusesEnum,
+  TransactionServerStatusesEnum
+} from 'types';
+import { chainIDSelector } from '../../reduxStore/selectors';
+import {
+  moveTransactionsToSignedState,
+  setNotificationModal,
+  setTransactionsDisplayInfo,
+  updateSignedTransactions
+} from '../../reduxStore/slices';
+import { store } from '../../reduxStore/store';
+
+export async function trackServerTransactions({
+  transactions,
+  transactionsDisplayInfo,
+  redirectAfterSign = true,
+  completedTransactionsDelay,
+  sessionInformation
+}: TrackTransactionsPropsType): Promise<SendTransactionReturnType> {
```

Comment:
since you are not awaiting anything inside this function, better to make it return a simple object not a promise